source/com.microsoft.tfs.core/src/com/microsoft/tfs/core/telemetry/TfsTelemetryHelper.java

```diff
@@ -75,22 +81,42 @@ public synchronized static TelemetryClient getTelemetryClient() {
     }
 
     public static void sendMetric(final String name, final double value) {
+        if (sendingTelemetryDisabled) {
```

Comment:
Just playing the devil's advocate:  would >0.5 seconds make any difference if we have left the initialization code as a static initializer?  I assume the jvm would take a while to startup, and can mask the time it takes to initialize this class?  So if we are going to mask the JVM start time, we probably should start the JVM before user tries to do their first action; and if we start JVM in advance, we can probably ignore this initialization time?    

On the other hand, the disabling logic is spread in quite a few places (and even in multiple classes as your modification shows), it's really not DRY.